GCEWindowsAgent/oslogin.go

```diff
@@ -0,0 +1,339 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+)
+
+var (
+	googleComment    = "# Added by Google Compute Engine OS Login."
+	googleBlockStart = "#### Google OS Login control. Do not edit this section. ####"
+	googleBlockEnd   = "#### End Google OS Login control section. ####"
+)
+
+type osloginMgr struct{}
+
+func (a *osloginMgr) diff() bool {
+	// True on first run.
+	return oldMetadata.Project.ProjectID == "" ||
+		// True if any value has changed.
+		(oldMetadata.Instance.Attributes.EnableOSLogin != newMetadata.Instance.Attributes.EnableOSLogin) ||
+		(oldMetadata.Instance.Attributes.TwoFactor != newMetadata.Instance.Attributes.TwoFactor) ||
+		(oldMetadata.Project.Attributes.EnableOSLogin != newMetadata.Project.Attributes.EnableOSLogin) ||
+		(oldMetadata.Project.Attributes.TwoFactor != newMetadata.Project.Attributes.TwoFactor)
+}
+
+func (a *osloginMgr) timeout() bool {
+	return false
+}
+
+func (a *osloginMgr) disabled(os string) bool {
+	return os == "windows"
+}
+
+func (a *osloginMgr) set() error {
+	oldenable := oldMetadata.Instance.Attributes.EnableOSLogin || oldMetadata.Project.Attributes.EnableOSLogin
```

Comment:
Use camelCase when you have two distinct words.

https://golang.org/doc/effective_go.html#mixed-caps